### PR TITLE
ENT-4444 Add health diagnostic API documentation

### PR DESCRIPTION
**Pull request by aleksandrychev**
ChangeLog: Title

Merge together: 
https://github.com/cfengine/mission-portal/pull/774
https://github.com/cfengine/nova/pull/1369

### CODE REVIEW

**Comment by aleksandrychev** [comment]
@nickanderson, please review

File: reference/enterprise-api-ref/health-diagnostic.markdown [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+```
+{
+    "hostsNeverCollected": 1,
+    "hostNotRecentlyCollected": 0,
+    "hostsUsingSameIdentity": 0,
+    "agentNotRunRecently": 2,
+    "lastAgentRunUnsuccessful": 0,
+    "totalFailed": 3,
+    "total": "50642"
```
Comment (nickanderson) [inline reference/enterprise-api-ref/health-diagnostic.markdown]: What does total include/not include (I'm thinking about hosts not reported)

File: reference/enterprise-api-ref/health-diagnostic.markdown [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
...timestamp as \"Last report collected\", agentstatus.lastagentlocalexecutiontimestamp as \"Last agent local execution time\"  \n                            FROM vm_hosts h \n                            LEFT JOIN agentstatus  ON agentstatus.Hostkey = h.Hostkey WHERE h.HostKey IN (SELECT result.hostkey FROM (SELECT agentstatus.HostKey \n                      FROM agentstatus \n                      LEFT JOIN vm_hosts ON vm_hosts.hostkey = agentstatus.hostkey\n                      WHERE  extract(epoch from (lastReportTimeStamp::timestamp - lastagentlocalexecutiontimestamp::timestamp)) > agentexecutioninterval::int  * 1.3) AS result \n                      WHERE hostkey IS NOT NULL  AND HostKey NOT IN (SELECT hostkey FROM hosts_not_reported) AND HostKey NOT IN (SELECT Hosts_view.HostKey \n                      FROM vm_hosts Hosts_view \n                      WHERE Hosts_view.lastreporttimestamp < to_timestamp('1549559891')) AND HostKey NOT IN (SELECT SameHosts.HostKey \n                      FROM (\n                          SELECT remotehostkey as HostKey FROM lastseenhostslogs GROUP BY remotehostkey HAVING COUNT(distinct remotehostip) > 1\n                          ) AS SameHosts))",
```
Comment (nickanderson) [inline reference/enterprise-api-ref/health-diagnostic.markdown]: Interesting. It returns the sql used to get it? Can this sql be used directly in the custom reporting ui?

Comment (aleksandrychev) [reply reference/enterprise-api-ref/health-diagnostic.markdown]: Yes, it returns the SQL as query API does. And yes this SQL can be used directly in the custom reporting UI.

File: reference/enterprise-api-ref/health-diagnostic.markdown (hunk 1 above)
Comment (aleksandrychev) [reply reference/enterprise-api-ref/health-diagnostic.markdown]: **total** is total host count, the same name was in Mission portal, so I didn't change it. Good note to rename this. I'll do it.

**Pull request merged**